subsys/net/ip/net_pkt.c

```diff
@@ -1966,9 +1966,7 @@ void *net_pkt_get_data(struct net_pkt *pkt,
 
 		return pkt->cursor.pos;
 	} else {
-		if (net_pkt_is_contiguous(pkt, access->size)) {
-			access->data = pkt->cursor.pos;
-		} else if (net_pkt_is_being_overwritten(pkt)) {
+		if (!net_pkt_is_contiguous(pkt, access->size)) {
```

Comment:
Could we create a test for this fix?

Reply:
Heh, I did not notice your comment about unit test :)

Reply:
a unit test would be definitely nice. It's been a while I put my nose in this API and can't recall why this was done like that (and thus why some tests are failing then)

Reply:
ah, know why... this patch is useless. 
overwrite is used to tell that a packet is now of fixed length. Thus why it is set an net_recv_data().
That said, tcp2 is suspiciously using it way too many times. Basically it should be used only when finalizing the packet there.

Reply:
> overwrite is used to tell that a packet is now of fixed length

Can we add this to the documentation, so that we don't have to dig it up again in the future?

Reply:
sure